### PR TITLE
Add gffread EXTRACT_CDNA and EXTRACT_CDS feature to outputs

### PR DESCRIPTION
**Pull request by liamlelievre**
As part of [#118](https://github.com/Plant-Food-Research-Open/genepal/issues/118), added additional Y.cdna.fasta and Y.cds.fasta outputs to the output dir annotations directory to the pipeline. These output files are useful for downstream tools like SNPEff.

## PR checklist

- [X] This comment contains a description of changes (with reason).
- [X] Make sure your code lints (`nf-core pipelines lint`).
- [X] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [X] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [X] Output Documentation in `docs/output.md` is updated.
- [X] `CHANGELOG.md` is updated.
- [X] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by GallVp** [comment]
Hi @liamlelievre 

This is amazing work.

We have used pre-commit to automate code linting. You can install and then enable it with

```bash
pip install pre-commit

cd genepal
pre-commit install
```

Then fix the errors by running the following twice,

```
pre-commit run --all-files
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 10. Added an HTML report [#44](https://github.com/Plant-Food-Research-Open/genepal/issues/44)
 11. Added content type as text/html for the MultiQC and genepal reports
 12. Added sra-tools for RNASeq data download [#102](https://github.com/Plant-Food-Research-Open/genepal/issues/102)
+13. Added cDNA and cds outputs to <OUTPUT_DIR>/annotations/<SAMPLE> directory [#118](https://github.com/Plant-Food-Research-Open/genepal/issues/118)
```
Comment (GallVp) [inline CHANGELOG.md:22]: Can you kindly start change log for a new version: 0.6.0?

0.5.0 has already been released.

File: docs/output.md [hunk 2]
```diff
@@ -168,7 +168,9 @@ If more than one genome is included in the pipeline, [ORTHOFINDER](https://githu
 - `annotations/`
   - `Y/`
     - `Y.gt.gff3`: Final annotation file for genome `Y` which contains gene models and their functional annotations
-    - `Y.pep.fasta`: Protein sequences for the gene models
+    - `Y.pep.fasta`: Protein sequences for the gene models\
```
Comment (GallVp) [inline docs/output.md:171]: The `\` is not needed.

File: subworkflows/local/gff_store.nf [hunk 3]
```diff
@@ -133,9 +135,34 @@ workflow GFF_STORE {
...
+        ch_cdna_extraction_inputs.map { meta, gff, fasta -> fasta}
+    )
+
+    ch_final_cdna                = EXTRACT_CDNA.out.gffread_fasta
+    ch_versions                  = ch_versions.mix(EXTRACT_CDNA.out.versions.first())
 
     emit:
     final_gff                   = ch_final_gff          // [ meta, gff ]
     final_proteins              = ch_final_proteins     // [ meta, fasta ]
+    final_cds                   = ch_final_cds          // [ meta, fasta ]
```
Comment (GallVp) [inline subworkflows/local/gff_store.nf:165]: Amazing!

**Comment by GallVp** [comment]
The tests will fail as the snapshots are old. You will have to install nf-test and then update snapshots with,

```
nf-test test --verbose --profile docker --updateSnapshot tests/minimal/main.nf.test
```

and

```
nf-test test --verbose --profile docker --updateSnapshot tests/stub/main.nf.test docker
```

File: conf/modules.config [hunk 4]
```diff
@@ -295,6 +295,27 @@ process { // SUBWORKFLOW: GFF_STORE
             saveAs: { filename -> filename.equals('versions.yml') ? null : filename },
         ]
     }
+
+    withName: '.*:GFF_STORE:EXTRACT_CDS' {
+        ext.args = '-x'
```
Comment (GallVp) [inline conf/modules.config:300]: Also here: `params.add_attrs_to_gffread_fastas`

Now that I think of it, `add_attrs_to_proteins_cds_fastas` is a better and more informative name. What do you think?

File: README.md [hunk 5]
```diff
@@ -139,6 +143,7 @@ The pipeline uses nf-core modules contributed by following authors:
 <a href="https://github.com/charles-plessy"><img src="https://github.com/charles-plessy.png" width="50" height="50"></a>
 <a href="https://github.com/bunop"><img src="https://github.com/bunop.png" width="50" height="50"></a>
 <a href="https://github.com/abhi18av"><img src="https://github.com/abhi18av.png" width="50" height="50"></a>
+<a href="https://github.com/liamlelievre"><img src="https://github.com/liamlelievre.png" width="50" height="50"></a>
```
Comment (GallVp) [inline README.md:146]: This section is automatically populated by parsing the `meta.yml` of modules/subworkflows. For example: https://github.com/Plant-Food-Research-Open/genepal/blob/ee702d71a9ea4f422c92729533ff0564c231b5e8/modules/nf-core/eggnogmapper/meta.yml#L82

Your name might get deleted unintentionally in a future update. That's why I have added it up top.

File: conf/modules.config (hunk 4 above)
Comment (liamlelievre) [reply conf/modules.config]: I agree, change made

File: README.md (hunk 5 above)
Comment (liamlelievre) [reply README.md:146]: Cool cheers

**Comment by GallVp** [comment]
Hi @liamlelievre 

I think all the issues have been resolved. To fix the pre-commit, I just had to rerun it on my system. I don't know why it was failing for you. Please feel free to merge the PR, once all the tests have passed!

Thank you again for your contribution and going through the hassle. Hopefully, the things you have picked up here will help you contribute to any nf-core pipeline in the future.

**Review by GallVp — APPROVED** [review]

**Pull request merged**